Yoast/YoastSEO.js
# Improve the snippet preview toggler

## Summary

Improve the snippet preview desktop and mobile views. Removes automatic switching of the views.

## Relevant technical choices:

On init, the desktop or mobile view mode is set based on the width of the snippet container. We can't use media queries since the available width depends on several factors such as the WP admin menu expanded/collapsed state or the 1/2 columns layout users can set in the WP Screen Options.

There's no more automatic switching to the mobile view.

The desktop view is always set to `overflow: auto` so it always works and it's up to browsers to show the horizontal scrollbar.

When the window gets resized or the WP admin menu state changes, the width calculation is still used but just to show or hide the "Scroll Hint" message. This part is optional, no objections if we prefer to remove it, just thought it was something useful for users.

Trying to summarize the changes:

- The metabox sidebar and the content sections use a new layout based on float, no more table layout: this is necessary because tables expand their width based on their content and we don't want that.
- Changed the snippet preview template to remove spaces between the slug elements
- When there's not enough space for the desktop view and so it's scrollable, a message "Scroll to see the preview content." gets displayed. Any thoughts for a better wording welcome.
- In the desktop view:
  - title, slug, and meta description have a fixed 600 pixels width
  - title and slug don't wrap in new lines and use ellipsis
  - meta description wraps in new lines
- In the mobile view:
  - title, slug, and meta description have a fluid width
  - slug doesn't wrap in new lines and doesn't have the "down arrow"
  - title and meta wrap in new lines, thus they don't have ellipsis


Note 1: in the mobile view, the title length check can't work correctly since the title wraps in new lines

Note 2: in the mobile view, the snippet elements width isn't 600 pixels anyways, because of the additional 16px + 16px padding

Note 3: as far as I see, in the mobile view Google truncates the url a bit before the max limit, not sure what we can do here.

Todo: evaluate if we want to add a short explanation in the inline help (i.e. the "?" button)

Todo: evaluate if we want to highlight the view switch in the UI, for example changing the "Snippet preview" heading in "Desktop snippet preview" and "Mobile snippet preview"

## Test instructions

- needs https://github.com/Yoast/wordpress-seo/pull/6700
- assets are already built, this should make testing easier, hopefully
- make sure YoastSEO.js is yarn linked
- on wordpress-seo, switch to the `stories/af/6682-snippet-preview-toggler` branch

Fixes #1071
